vernac/comInductive.ml

```diff
@@ -491,11 +492,12 @@ let interp_mutual_inductive_constr ~sigma ~template ~udecl ~variances ~ctx_param
       constructors
   in
   let sigma, (default_dep_elim, arities) = inductive_levels env_ar_params sigma arities ctor_args in
-  let sigma = Evd.minimize_universes sigma in
+  let lbound = if poly then UGraph.Bound.Set else UGraph.Bound.Prop in
+  let sigma = Evd.minimize_universes ~lbound sigma in
```

Comment:
Having to to this is IMO an indication that we're not doing the right thing.

If we naively always pass Set a problem appears with 
~~~coq
Inductive T A := C : A -> T nat -> T A.
~~~
(a version with `option A` instead of `nat` appears in the test suite)
before minimization we have generated `T : Type@{Type|u} -> Type@{q|v}` with `Set <= u, u <= v`
(`Set <= u` from the application `T nat`, `u <= v` from `include_constructor_argument`)
Minimization collapses `q := Type` and sets `u = v`, with lbound Set it also forgets `Set <= u`. We then think the inductive can be template and the kernel will reject it.

Probably we need to
- in `include_constructor_argument` in monomorphic mode do not ignore qualities, such that `Set <= Type@{q|u}` produces `q := Type` instead of leaving it floating, this would make the example generate `q := Type`. (`include_constructor_argument` currently has a comment that ignoring qualities only impacts squashing but that's not correct once we use sort qualities in monomorphic mode)
- when determining if an inductive may be template, the parameters binding a universe must be at floating (possibly above_prop) quality (this is an interpretation of unbounded_from_below now that we can't check the presence of `Set <= u` constraints)
- to be able to do this check we need to delay sort quality collapse (currently done in minimize_universes)

then we should be able to forget `Set <= u` constraints in minimization ie always use lbound = Set (and the lbound concept can then be deleted since this is the last user)